apps/dashboard/app/assets/javascripts/application.js

```diff
@@ -71,3 +72,63 @@ function installSettingHandlers(name) {
   var selector = 'input[type="range"][name="' + name + '"]';
   $(selector).change(storeSetting);
 }
+
+
+function promiseLoginToXDMoD(xdmodUrl){
+  return new Promise(function(resolve, reject){
+
+    var promise_to_receive_message_from_iframe = new Promise(function(resolve, reject){
+      window.addEventListener("message", function(event){
+        if (event.origin !== xdmodUrl){
+          console.log('Received message from untrusted origin, discarding');
+          return;
+        }
+        else if(event.data.application == 'xdmod'){
+          if(event.data.action == 'loginComplete'){
+            resolve();
+          }
+            else if(event.data.action == 'error'){
+              console.log('ERROR: ' + event.data.info);
+              //FIXME: what to pass to reject?
+              reject();
```

Comment:
pass event.data.info if it is available, and otherwise, an Error object through this reject